src/server.rs

```diff
@@ -378,12 +378,10 @@ pub extern "C" fn rustls_server_session_read(
         let out_n: &mut size_t = try_ref_from_ptr!(out_n, &mut size_t);
         let n_read: usize = match session.read(read_buf) {
             Ok(n) => n,
-            // The CloseNotify TLS alert is benign, but rustls returns it as an Error. See comment on
-            // https://docs.rs/rustls/0.19.0/rustls/struct.ServerSession.html#impl-Read.
-            // Log it and return EOF.
+            // Rustls turns close_notify alerts into `io::Error` of kind `ConnectionAborted`.
+            // https://docs.rs/rustls/0.19.0/rustls/struct.ClientSession.html#impl-Read.
             Err(e) if e.kind() == ConnectionAborted && e.to_string().contains("CloseNotify") => {
```

Comment:
Could you factor this predicate into a function `is_close_notify(error: std::io::Error) -> bool` that could be shared between `server.rs` and `client.rs`? Or maybe provide `impl From<io::Error> for rustls_result` so you can just do
```
match result {
    Ok(n) => n,
    Err(e) => return e.into();
};
```